packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1990,6 +1990,7 @@ export class ContainerRuntime
 	public orderSequentially<T>(callback: () => T): T {
 		let checkpoint: IBatchCheckpoint | undefined;
 		let result: T;
+		this.flush();
```

Comment:
This does not look right. You can't flush in the middle of the batch (we support reentrancy for orderSequentially).
You can either have enough state to ignore already scheduled flush micro-task (need some counter) OR/and not schedule micro-task for ops within orderSequentially flow. I.e. the rest of changes minus this change.

Reply:
One of my old tests caught this! 😄